From 6a6eca0c9604a9bd508d98b75c5608f20a3a7bf6 Mon Sep 17 00:00:00 2001
From: Vojtech Trefny <vtrefny@redhat.com>
Date: Thu, 24 Oct 2024 12:18:58 +0200
Subject: [PATCH] Align sizes up for growable LVs

Growable LVs usually start at minimum size so adjusting it down
can change the size below allowed minimum.

Resolves: RHEL-8036
Resolves: RHEL-19725
---
 blivet/devices/lvm.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/blivet/devices/lvm.py b/blivet/devices/lvm.py
index 62974443..1293cae2 100644
--- a/blivet/devices/lvm.py
+++ b/blivet/devices/lvm.py
@@ -2574,7 +2574,7 @@ class LVMLogicalVolumeDevice(LVMLogicalVolumeBase, LVMInternalLogicalVolumeMixin
         if not isinstance(newsize, Size):
             raise ValueError("new size must be of type Size")

-        newsize = self.vg.align(newsize)
+        newsize = self.vg.align(newsize, roundup=self.growable)
         log.debug("trying to set lv %s size to %s", self.name, newsize)
         # Don't refuse to set size if we think there's not enough space in the
         # VG for an existing LV, since it's existence proves there is enough
--
2.47.0

